doc/plutus-core-spec/cardano/cardano.tex

```diff
@@ -31,7 +36,48 @@ \section{Built-in types and functions}
     \label{table:bs-variants}
 \end{table}
 
+
+\paragraph{Concrete syntax for built-in types.}
+Recall that in the abstract notation for built-in types introduced in
+Section~\ref{sec:built-in-types}, a built-in type is either an \textit{atomic
+  type} such as \texttt{integer} or \texttt{string} or an application $\op(T_1,
+\ldots, T_n)$ of a \textit{type operator} to a sequence of built-in types.  The
+concrete syntax of built-in types used in textual Plutus Core programs is
+slightly different in that we use a curried form of application for type
+operators: a type is given by
+
+\newcommand{\bitn}{\mathbf{T}} % built-in type name
+\newcommand{\bitc}{\mathbf{C}} % built-in constant syntax
+
+\begin{minipage}{\linewidth}
+    \centering
+    \[\begin{array}{rclr}
+    \bitn & ::= & \textit{atomic-type} & \textrm{Atomic type}\\
+                & & \texttt{(}\op \ \bitn_1 \ldots \bitn_{\valency{\op}}) & \textrm{Type application}\\
+    \end{array}\]
+    \label{fig:built-in-type-concrete-syntax}
+\end{minipage}
+\noindent Note that we again require that all type operators are fully applied.
+We refer to the syntactic objects $\bitn$ above as \textit{concrete built-in
+ types}. There is an obvious bijection between these and the abstract built-in
+types used elsewhere in this document, and given an abstract built-in type $T$
+we will denote the corresponding concrete built-in type by $\bar{T}$.
+
```

Comment:
^ The fact that the concrete syntax for polymorphic types differs from the notation that we use everywhere else in the specification is a bit annoying.  Maybe we should use the curried notation everywhere (although that suggests that partial application might be possible, which I sincerely hope we'll never allow).  Changing that would have required a lot of rewriting and would have been easy to get wrong, so for the time being I've just added a fudge here.

Reply:
I think this is totally fine. One thing is the concrete syntax, and another the abstract one. We don't need to conflate them.

Reply:
Why does the abstract syntax need to use the uncurried form?